src/__experimental__/components/date-range/date-range.d.ts

```diff
@@ -1,17 +1,43 @@
 import * as React from "react";
-import DateInput from "../date";
+import { DateInputProps } from "../date/date";
+import { TextboxProps } from "../textbox/textbox";
 
-export interface DateRangeProps {
+export interface DateRangeProps extends TextboxProps {
+  /* The default value of the input if it's meant to be used as an uncontrolled component */
+  defaultValue?: string[];
+  /** Props for the child end Date component */
+  endDateProps?: DateInputProps;
   /** Optional label for endDate field */
   endLabel?: string;
+  /** Indicate that error has occurred on end date
+   * Pass string to display icon, tooltip and red border
+   * Pass true boolean to only display red border
+   */
+  endError?: boolean | string;
+  /** Indicate additional information for end date
+   * Pass string to display icon, tooltip and blue border
+   * Pass true boolean to only display blue border
+   */
+  endInfo?: boolean | string;
+  /** Indicate that warning has occurred on end date
+   * Pass string to display icon, tooltip and orange border
+   * Pass true boolean to only display orange border
+   */
+  endWarning?: boolean | string;
+  /** An optional string prop to provide an id to the component */
+  id?: string;
+  /** Display labels inline */
+  labelsInline?: boolean;
+  /** An optional string prop to provide a name to the component */
+  name?: string;
   /** Specify a callback triggered on change */
   onChange?: (ev: React.ChangeEvent<HTMLElement>) => void;
```

Comment:
This typing is incorrect, this callback will always populate `input.value` with an array. In order to avoid failures further down the line (eg. forcing tests that do not meet Carbon's behaviour and therefore need to mock this behaviour) please type this correctly.

Type should be `(ev: React.ChangeEvent<CustomCarbonType>) => void` where `CustomCarbonType` contains mandatory value as an array if I am not wrong.

Reply:
I agree, we'll need our own type because `value` is an object.

Reply:
looks like there is a custom event object in the DateRange passed to the `onChange` prop. I'll create a new interface for it that you could use.